commit e71a82d67fd0b762d98b94a195f6f34a487c1182 (refs/changes/86/2486/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-08-31T16:36:48-07:00 (2 years, 1 month ago)
    
    NET-376 hn-netfile should work for a server that isn't yet a headnode

diff --git a/bin/hn-netfile b/bin/hn-netfile
index 9931490..c0e711c 100755
--- a/bin/hn-netfile
+++ b/bin/hn-netfile
@@ -118,37 +118,37 @@ function getHeadnodeInfo(opts, callback) {
 
     var i;
 
-    opts.cnapi.listServers({headnode: true}, function (err, headnodes) {
+    opts.cnapi.listServers(function (err, servers) {
         if (err) {
             callback(err);
             return;
         }
 
         // Try `uuid` match first ...
-        var headnode;
-        for (i = 0; i < headnodes.length; i++) {
-            if (headnodes[i].uuid === opts.headnodeHostnameOrUuid) {
-                headnode = headnodes[i];
+        var server;
+        for (i = 0; i < servers.length; i++) {
+            if (servers[i].uuid === opts.headnodeHostnameOrUuid) {
+                server = servers[i];
                 break;
             }
         }
 
         // ... then `hostname`.
-        if (!headnode) {
-            for (i = 0; i < headnodes.length; i++) {
-                if (headnodes[i].hostname === opts.headnodeHostnameOrUuid) {
-                    headnode = headnodes[i];
+        if (!server) {
+            for (i = 0; i < servers.length; i++) {
+                if (servers[i].hostname === opts.headnodeHostnameOrUuid) {
+                    server = servers[i];
                     break;
                 }
             }
         }
 
-        if (headnode) {
-            opts.cn_uuid = headnode.uuid;
-            opts.hostname = headnode.hostname;
+        if (server) {
+            opts.cn_uuid = server.uuid;
+            opts.hostname = server.hostname;
             callback();
         } else {
-            callback(new Error('Could not find a headnode=true server "'
+            callback(new Error('Could not find server "'
                 + opts.headnodeHostnameOrUuid + '" in CNAPI'));
         }
     });
